packages/components/app/styles/components/dropdown.scss

```diff
@@ -110,37 +110,64 @@ $hds-dropdown-toggle-border-radius: 5px;
 // UL ELEMENT
 // GOES INSIDE HDS::DISCLOSURE's :content block
 
-.hds-dropdown-list {
+.hds-dropdown-content {
+  display: flex;
+  flex-direction: column;
   width: max-content; // notice: this is important because being in a position absolute means the layout algorithm assigns a width of 0 and this impacts on the flex algorithm of the children (in some cases they don't use the full width)
   min-width: 200px;
   max-width: 400px;
-  margin: 0;
-  padding: 4px 0;
-  list-style: none;
   background-color: var(--token-color-surface-primary);
   border-radius: 6px;
   box-shadow: var(--token-surface-high-box-shadow);
 }
 
-.hds-dropdown-list--fixed-width {
+.hds-dropdown-content--fixed-width {
   min-width: initial;
   max-width: initial;
 }
 
-.hds-dropdown-list--position-right {
+.hds-dropdown-content--position-right {
   position: absolute;
   top: calc(100% + 4px);
   right: 0;
   z-index: 2; // https://github.com/hashicorp/design-system/issues/114
 }
 
-.hds-dropdown-list--position-left {
+.hds-dropdown-content--position-left {
   position: absolute;
   top: calc(100% + 4px);
   left: 0;
   z-index: 2; // https://github.com/hashicorp/design-system/issues/114
 }
 
+.hds-dropdown-header {
+  flex: none;
+  padding: 6px 8px;
+}
+
+.hds-dropdown-list {
+  flex: 1 1 auto;
+  margin: 0;
+  padding: 4px 0;
+  overflow-y: auto;
+  list-style: none;
+  overscroll-behavior: contain;
+}
+
+.hds-dropdown-footer {
+  position: relative;
+  flex: none;
+  padding: 10px 8px 6px;
+
+  &::after {
```

Comment:
I don't think this should be implemented this way. My understanding, looking at the Figma designs, is that the line above the content of the footer is a `separator` (though, with a different size).

Plus, I suggest to review and discuss the Figma implementation with @heatherlarsen (or @jorytindall if there's already been an handover) because in Figma they're using a `Generic item` to mimic the footer (and maybe the header too) but they should be separate generic containers, with their own padding specifications to avoid misalignments between code and design.

Reply:
For reference: https://www.figma.com/file/U5cpVPoELaUktOex3ByG1j?node-id=0:1#375922763

(plus, it came to mind that the separator internally uses an `<li>` element, so... we have to find a way to or use it directly (with a different tag) or make its styling shareable).

(maybe we could consider an optional `@hasSeparator` argument for the `header/footer` elements?)

Reply:
I remember discussing this with @heatherlarsen last week, making the point that in code using a `list-item/separator` doesn't make sense semantically, and a simple border makes more sense (which is the current implementation). Not sure how feasible it is for this to be done in Figma though

Reply:
@didoo, I've added the `@hasSeparator` argument, as suggested, so this should be good for another review